### ISSUE TITLE
Why does `Int32 === Int32` gives false?

### ISSUE THREAD

**Issue by bew** [crystal-lang/crystal#4362]
I have a class in a variable, I'm trying to avoid `if ...; elsif ...; else ...; end` branching:
```cr
klass = Int32
case klass
when Int32
  p true
else
  p "wtf"
end
```
This prints `"wtf"`...
https://carc.in/#/r/1yfy

**Comment by luislavena** [comment]
Hello,

Here is the code that triggers that:

https://github.com/crystal-lang/crystal/blob/master/src/class.cr#L14-L16

`Int32` is a `Class`, so `Class#===` checks if `other` is an instance of `self` (in this case, `Int32`) using `is_a?` which checks instances, not classes.

```crystal
klass = Int32
pp (Int32 == klass)  # => true
pp (Int32 === klass) # => false
pp (Int32 === 1)     # => true
```

In the documentation for `case`, this scenario is described:

https://crystal-lang.org/docs/syntax_and_semantics/case.html

> Note that === is used for comparing an expression against a case's value.
>
> If a when's expression is a type, is_a? is used. Additionally, if the case expression is a variable or a variable assignment the type of the variable is restricted

**Comment by kirbyfan64** [comment]
**TL;DR:** [Use `.class` in your `when` branch:](https://carc.in/#/r/1yfz)

```crystal
klass = Int32
case klass
when Int32.class
  p true
else
  p "wtf"
end
```

**Comment by luislavena** [comment]
Thanks @kirbyfan64, hit *Comment* too early without completing the comment :smile:

**Comment by bew** [comment]
@luislavena I know the concret "why", I'm asking about the "why this decision has been taken" (I agree, it wasn't crystal clear 😄 )

@kirbyfan64 thanks! Now I don't get why this workaround works:
```cr
klass = Int32
pp klass === Int32.class # => false
pp klass.class # => Class
```
It doesn't work when checking manually.. https://carc.in/#/r/1yg1
So why does it work in case expr?

**Comment by c910335** [comment]
https://github.com/crystal-lang/crystal/blob/master/src/class.cr#L14
```crystal
# in class Class
def ===(other)
  other.is_a?(self)
end
```
so
```crystal
klass = Int32
pp Int32.class === klass # => true
pp klass.is_a?(Int32.class) # => true
```

**Comment by c910335** [comment]
`klass === Int32.class` is different from `Int32.class === klass`

**Comment by bew** [comment]
Okayyy, thanks @c910335 !

**Comment by ozra** [comment]
It is quiet counter intuitive "help", not matching the _actual value_ (the type), akin to crossing the river to get water.
Wouldn't it be more reasonable to at least:
```
def ===(other)
  other.is_a?(self) || other == self
end
```
